### PR TITLE
Various Modifications

### PR DESCRIPTION
**Pull request by johnwarden**
Hey,

So this package was a great starting point for me working on a mtgox daytrade app.  I ended up making a bunch of modifications to it as I encountered various challenges.

The modifications include most importantly some reconnection logic, because I found mtgox was constantly disconnecting mysteriously, and for some reason the socket.io library wasn't able to reconnect and just ended up hanging.  This code will detect that scenario and automatically kill and recreate the client.

I have also added the ability for the user to optionally choose which channels to subscribe to, as well as the optional ability to subscribe to client-specific channels (e.g. my individual trades) if the user includes a keys.js file.

Also made misc changes that made this module more useable in the Npm/Meteor environment I was using.

I'm not sure that all these changes make sense to you, especially some backwords-compatibility-breaking changes.  But take a look and tell me what you think!

### CODE REVIEW

File: mtgox.js [hunk 1]
```diff
@@ -1,98 +1,231 @@
...
+		{ "key":"85174711-be64-4de1-b783-0628995d7914", "private":"lag", "name":"Lag" }
+];
+
+
+/* Look up channel by name (e.g. "Market Depth", id (e.g. 24e67e0d-1cad-4cc0-9e7a-f8523ef460fe) or type (e.g. "depth") */
 
 var getChannel = function(key) {
   return MTGOX_CHANNELS.filter(function(channel) {
-    return (channel.key == key || channel.private == key);
+    return (channel.key == key || channel.private == key || channel.key == key);
```
Comment (olalonde) [inline mtgox.js]: Why is there 2 `channel.key == key`? Could you remove this please!

**Comment by olalonde** [comment]
Thanks for this! I haven't tested it because I don't have time but I will trust you did ;) Please just make the edit I suggested before I merge.

File: mtgox.js (hunk 1 above)
Comment (johnwarden) [inline mtgox.js]: Will do.

**Comment by olalonde** [comment]
Ok, I published under `0.1.0` in npm.

**Comment by johnwarden** [comment]
cool!

**Pull request merged**